### ISSUE TITLE
"opam update" fails in "patch" on Cygwin

### ISSUE THREAD

**Issue by QinshiWang** [ocaml/opam#4659]
When I run `opam update` or `opam update default`, I get
```
[default] synchronised from git+https://github.com/fdopen/opam-repository-mingw.
git#opam2
[ERROR] Could not update repository "default": "C:\\OCaml64\\bin\\patch.exe -p1
        -i /home/s-quark/.opam/log/processed-patch-7092-42bb52" exited with
        code 1
```
It shows some error happens when applying the patch to my local repository list.
```
# opam config report
# opam-version      2.0.7
# self-upgrade      no
# system            arch=x86_64 os=win32 os-distribution=cygwinports os-version=
10.0.19041
# solver            builtin-mccs+glpk
# install-criteria  -removed,-count[version-lag,request],-count[version-lag,chan
ged],-changed
# upgrade-criteria  -removed,-count[version-lag,solution],-new
# jobs              15
# repositories      1 (http), 1 (version-controlled)
# pinned            0
# current-switch    4.10.0+mingw64c
```

**Comment by dra27** [comment]
Would you be able to attach (or upload and reference) the patch?

**Comment by QinshiWang** [comment]
Thanks. I looked into the patch****.out file and find
```
patching file packages/conf-gmp/conf-gmp.1/conf-gmp.1/files/test.o
Not deleting file packages/conf-gmp/conf-gmp.1/conf-gmp.1/files/test.o as content differs from patch
```
That is because I tried to resolve a compatibility issue by editing files in OPAM. So my problem should be how to restore a distorted OPAM local repo?

**Comment by dra27** [comment]
I'm not completely clear how you got to this state - is this is a local repository which you've added with `opam repo add` or have you been directly using files in `.opam/repo` ?! Either way, I think if you delete the `test.o` files (wherever they are!), you should unblock things.

You should _never_ try to edit files in the `repo` directory of the opam root - the workflow for dealing with a package problem is to clone the repository locally and then add that git clone with `opam repo add`

**Comment by QinshiWang** [comment]
Thanks. I fixed it by `opam repo remove default --all` (`--all` is needed).

Detailed story: I looked into `.opam` and find the `test.o` is already deleted, and that's the exact reason why it finds the file differs (existing versus deleted). That should because I changed/deleted the file a few months ago. I think it's because OPAM keeps a hash value somewhere but I don't know. When executing `update`, OPAM sends the hash value to the remote source to indicate the local version and the remote answers with a patch.
